thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/MultiThreadView.kt

```diff
@@ -169,81 +175,66 @@ class MultiThreadView(
         }
     }
 
-    // Details pane replaced with comparison pane
     private var comparison = ThreadComparisonPane(threadDumps.size, threadDumps[0].version)
 
-    private var allThreads: List<Thread> = threadDumps.flatMap { it.threads }
-
-    private var poolList = PoolList(allThreads.groupingBy(Thread::pool).eachCount())
-    private var systemList = SystemList(allThreads.groupingBy(Thread::system).eachCount())
-    private var stateList = StateList(allThreads.groupingBy(Thread::state).eachCount())
+    private val poolList = FilterList("(No Pool)")
+    private val systemList = FilterList("Unassigned")
+    private val stateList = FilterList("")
     private val threadDumpCheckboxList = ThreadDumpCheckboxList(paths)
     private var listModelsAdjusting = false
 
     private val exportButton = JMenuBar().apply {
-        val fileName = "threaddump_${threadDumps[0].version}_${threadDumps[0].hashCode()}"
+        val firstThreadDump = threadDumps.first()
+        val fileName = "threaddump_${firstThreadDump.version}_${firstThreadDump.hashCode()}"
         add(exportMenu(fileName) { mainTable.model })
         isVisible = mainTable.model.isSingleContext
     }
 
     private val searchField = JXSearchField("Search")
 
-    private val filters: List<(thread: Thread) -> Boolean> = buildList {
-        add { thread ->
-            thread.state in stateList.checkBoxListSelectedValues
+    private fun filter(thread: Thread?): Boolean {
```

Comment:
The list-of-predicates approach was cool, but I suspect slower than necessary. This has the same semantics, and while slightly harder to reason about, should be a good bit faster